web/cypress/integration/route/create-route-with-service-discovery-upstream.spec.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-undef */
+
+context('Create Route with Service Discovery Upstream', () => {
+  const selector = {
+    name: '#name',
+    description: '#desc',
+    discovery_type: '#discovery_type',
+    service_name: '#service_name',
+    upstreamSelector: '[data-cy=upstream_selector]',
+    nameSelector: '[title=Name]',
+    selectItem: '.ant-select-item-option-content',
+    drawer: '.ant-drawer-content',
+    monacoScroll: '.monaco-scrollable-element',
+    input: ':input',
+    deleteAlert: '.ant-modal-body',
+    notification: '.ant-notification-notice-message',
+  };
+
+  const data = {
+    deleteRouteSuccess: 'Delete Route Successfully',
+    submitSuccess: 'Submit Successfully',
+    deleteUpstreamSuccess: 'Delete Upstream Successfully',
+    weight: 1,
+    description: 'desc_by_autotest',
+    upstreamName: 'test_upstream',
+    routeName: 'test_route',
+    serviceName: 'test.cluster.local',
+  };
+
+  beforeEach(() => {
+    cy.login();
+  });
+
+  it('should create route with DNS service discovery upstream', function () {
+    cy.visit('/');
+    cy.get('[role=menu]')
+      .should('be.visible')
+      .within(() => {
+        cy.contains('Route').click();
+      });
+    cy.contains('Create').click();
+
+    cy.contains('Next').click().click();
+    cy.get(selector.name).type(data.routeName);
+    cy.contains('Next').click();
+
+    // should enable Upstream input boxes after selecting Custom mode
+    cy.get(selector.upstreamSelector).click();
+    cy.contains('.ant-select-item-option-content', 'Custom').click();
+
+    // set service discovery
+    cy.get('[title="Node"]').click();
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('Service Discovery').click();
+    });
+
+    cy.get(selector.discovery_type).click();
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('DNS').click();
+    });
+    cy.get(selector.service_name).type(data.serviceName);
```

Comment:
last one suggestion about your test case:

add an assertion to confirm that there are not `discovery_args` and `namespace_id` when choose `DNS`

Reply:
Here, I do this assert in `upstream` e2e test.
`web/cypress/integration/upstream/create_and_delete_upstream.spec.js`